nominal/_utils.py

```diff
@@ -198,12 +207,20 @@ def _flexible_time_to_conjure_scout_run_api(
         return scout_run_api.UtcTimestamp(seconds_since_epoch=seconds, offset_nanoseconds=nanos)
     raise TypeError(f"expected {datetime} or {IntegralNanosecondsUTC}, got {type(timestamp)}")
 
+def _datetime_to_conjure_datasource_api(timestamp: datetime) -> datasource.Timestamp:
+    seconds, nanos = _datetime_to_seconds_nanos(timestamp)
+    return datasource.Timestamp(seconds=seconds, nanos=nanos)
 
 def _conjure_time_to_integral_nanoseconds(
     ts: scout_run_api.UtcTimestamp,
 ) -> IntegralNanosecondsUTC:
     return ts.seconds_since_epoch * 1_000_000_000 + (ts.offset_nanoseconds or 0)
 
+def _datasource_api_timestamp_to_datetime(
+    ts:  datasource.Timestamp
+) -> datetime:
+    seconds, nanos = ts.seconds, ts.nanos
+    return datetime.fromtimestamp(seconds, tz=timezone.utc) + timedelta(microseconds=nanos // 1000)
```

Comment:
`datetime` doesn't support nanoseconds, should I just use `IntegralNanosecondsUTC`?

Reply:
`datetime` also doesn't make much sense with relative timestamps if there are already logs in relative time..